### DISCUSSION TITLE
Is it possible to filter music by given fields within the current specification?

### DISCUSSION THREAD

**Discussion by tsshadow** [opensubsonic/open-subsonic-api#82] [General]
Hey everyone,

First of all, i tried to make my situation as clear as possible, please bear with me.

I have been using LMS (which supports open-subsonic) for a while now, but my use-case was always missing. 
Therefore i had to patch the subsonic API for LMS.

Since LMs uses the open subsonic API I would like to discuss my use case, and whether we could think of a way to implement it within the open subsonic borders ( or maybe think of an API extension). 

My use case is as follows. 

I have a lot of music which is categorized in `genre`, `subgenre`, `year` and `rating`.
I categorize my music with mediamonkey.
I (mis)use the `mood` tag for `subgenre`. 

This makes it possible for me to create a playlist on the fly with these fields.  

I currently use `/getMoods`  to return a list of `moods`
and i use `/getSongsByMood`  to fetch me a list of songs. 

`/getMoods` is a sort of copy of `/getGenres` 
`/getSongsByMood` is a sort of copy of `/getSongsByGenre` 

Currently `/getSongsByMood` has the following required parameters: `mood` 
The following are optional: `count`, `year`, `ratingMin`, `ratingMax` 

Currently `/getSongsByGenre` has the following required parameters: `genre` 
The following are optional: `count`, `year`, `ratingMin`, `ratingMax` 

What i try to achieve is that i can get a list of songs with several filters. 
(and also fetch the filter values, like with getMoods etc.)

As far as i know, this is not currently possible in the current opensubsonic spec. 
What are your views on this usecase?

**Comment by epoupon** [comment]
Hello!
Thanks for your post. Needs are actually gathered here: https://github.com/opensubsonic/open-subsonic-api/discussions/57
IMO, combining multiples filters (AND) is quite straightforward and should be easy to implement for all servers.
